File: packages/core/src/theme.ts
/** Per-theme tokens: colors for canvas, nodes, text, flows. */
export interface Theme {
  canvas: string;
  dot: string;
  nodeFill: string;
  nodeStroke: string;
  title: string;
  subtitle: string;
  nodeText: string;
  bodyText: string;
  label: string;
  edge: string;
  chipStroke: string;
  flows: Record<'sky' | 'emerald' | 'amber' | 'rose', string>;
  packet: string;
}

export const dark: Theme = {
  canvas: '#0B1118',
  dot: '#1B2737',
  nodeFill: '#0E1620',
  nodeStroke: '#22304A',
  title: '#E2E8F0',
  subtitle: '#64748B',
  nodeText: '#E2E8F0',
  bodyText: '#94A3B8',
  label: '#64748B',
  edge: '#1F2B3D',
  chipStroke: '#38BDF8',
  flows: { sky: '#38BDF8', emerald: '#34D399', amber: '#F59E0B', rose: '#F87171' },
  packet: '#7DD3FC',
};

export const light: Theme = {
  canvas: '#F8FAFC',
  dot: '#E2E8F0',
  nodeFill: '#FFFFFF',
  nodeStroke: '#CBD5E1',
  title: '#0F172A',
  subtitle: '#64748B',
  nodeText: '#0F172A',
  bodyText: '#475569',
  label: '#64748B',
  edge: '#CBD5E1',
  chipStroke: '#0284C7',
  flows: { sky: '#0284C7', emerald: '#059669', amber: '#D97706', rose: '#DC2626' },
  packet: '#0EA5E9',
};

File: dotnet/src/FlowInk.Core/Theme.cs
namespace FlowInk.Core;

/// <summary>Theme tokens — mirrors theme.ts. Dark is the showcase default.</summary>
public sealed record Theme(
    string Canvas,
    string Dot,
    string NodeFill,
    string NodeStroke,
    string Title,
    string Subtitle,
    string NodeText,
    string BodyText,
    string Label,
    string Edge,
    string ChipStroke,
    string Sky,
    string Emerald,
    string Amber,
    string Rose,
    string Packet)
{
    public string Flow(string name) => name switch
    {
        "emerald" => Emerald,
        "amber" => Amber,
        "rose" => Rose,
        _ => Sky,
    };

    public static Theme Dark { get; } = new(
        Canvas: "#0B1118", Dot: "#1B2737", NodeFill: "#0E1620", NodeStroke: "#22304A",
        Title: "#E2E8F0", Subtitle: "#64748B", NodeText: "#E2E8F0", BodyText: "#94A3B8",
        Label: "#64748B", Edge: "#1F2B3D", ChipStroke: "#38BDF8",
        Sky: "#38BDF8", Emerald: "#34D399", Amber: "#F59E0B", Rose: "#F87171",
        Packet: "#7DD3FC");

    public static Theme Light { get; } = new(
        Canvas: "#F8FAFC", Dot: "#E2E8F0", NodeFill: "#FFFFFF", NodeStroke: "#CBD5E1",
        Title: "#0F172A", Subtitle: "#64748B", NodeText: "#0F172A", BodyText: "#475569",
        Label: "#64748B", Edge: "#CBD5E1", ChipStroke: "#0284C7",
        Sky: "#0284C7", Emerald: "#059669", Amber: "#D97706", Rose: "#DC2626",
        Packet: "#0EA5E9");
}
